app/services/checksum_validator.rb

```diff
@@ -143,38 +140,30 @@ def validate_signature_catalog_entry(entry)
     results.add_result(AuditResults::FILE_NOT_IN_MOAB, absent_from_moab_data)
   end
 
+  # @return [String]
   def signature_catalog_entry_path(entry)
     @signature_catalog_entry_paths ||= {}
     @signature_catalog_entry_paths[entry] ||= "#{object_dir}/#{entry.storage_path}"
   end
 
+  # @return [String]
   def latest_signature_catalog_path
-    @latest_signature_catalog_path ||= latest_moab_version.version_pathname.join(MANIFESTS, SIGNATURE_XML).to_s
+    @latest_signature_catalog_path ||= latest_moab_version.version_pathname.join(MANIFESTS, 'signatureCatalog.xml').to_s
   end
 
-  # shameless green implementation
+  # @return [Array<SignatureCatalogEntry>]
   def latest_signature_catalog_entries
-    @latest_signature_catalog_entries ||= begin
-      if latest_moab_version.signature_catalog
-        latest_moab_version.signature_catalog.entries
-      else
-        absent_from_moab_data = { signature_catalog_path: latest_moab_version.signature_catalog }
```

Comment:
Note the hash tries to use the value it just tested in the conditional as being falsey (likely nil)